lib/msf/core/exploit/remote/postgres.rb

```diff
@@ -143,16 +147,17 @@ def postgres_logout
   # @param sql [String] The query to run
   # @param doprint [Boolean] Whether the result should be printed
   # @return [Hash]
-  def postgres_query(sql=nil,doprint=false)
-    ip = datastore['RHOST']
-    port = datastore['RPORT']
-    postgres_login unless self.postgres_conn
+  def postgres_query(sql = nil, doprint = false)
     unless self.postgres_conn
-      return {:conn_error => true}
+      result = postgres_login
+      unless result == :connected
+        return { :conn_error => result }
+      end
     end
+
     if self.postgres_conn
       sql ||= datastore['SQL']
-      vprint_status "#{ip}:#{port} Postgres - querying with '#{sql}'"
+      vprint_status "#{rhost}:#{rport} Postgres - querying with '#{sql}'"
```

Comment:
I think you want to leave these as `ip` and `port` no? 
you have this above:
```
    ip = opts[:server]         || rhost
    port = opts[:port]         || rport
```

Reply:
After having a look at this, I think this should be rhost and rport. `ip` and `port` are defined outside the scope of this function.